src/darwin/Framework/CHIP/MTROTAImageTransferHandler.mm

```diff
@@ -28,9 +28,16 @@
 
 constexpr uint32_t kMaxBdxBlockSize = 1024;
 
+constexpr double kMilliSecondsInSecond = 1000.0;
+
 // Timeout for the BDX transfer session. The OTA Spec mandates this should be >= 5 minutes.
 constexpr System::Clock::Timeout kBdxTimeout = System::Clock::Seconds16(5 * 60);
 
+// For thread devices, we need to throttle sending Blocks in response to BlockQuery messages
+// to avoid spamming the network with too many BDX messages. We are going to match the polling
+// interval of 50 ms as the time to wait before sending a Block in response to a BlockQuery.
+constexpr System::Clock::Timeout kBdxThrottleIntervalInMsecs = System::Clock::Milliseconds32(50);
```

Comment:
Oh, and: 
```suggestion
constexpr auto kBdxThrottleInterval = System::Clock::Milliseconds32(50);
```

If you use the typesafe time types throughout, you don't need to label your units in the variable names, because the type knows the units and you can't mess that part up.  And if you need to do `count` at that point cast to the right type as below.